CDC-Data-Reconciliation-Backend/server.py

```diff
@@ -63,6 +64,30 @@ async def manual_report(state_file: UploadFile = File(None), cdc_file:  UploadFi
     return res
 
 
+@app.get("/reports/{report_id}")
+async def get_report_summary(report_id: int):
+    """
+    Endpoint to fetch a report by its ID.
+    """
+    report = fetch_report_from_db(report_id)
+    if report is None:
+        raise HTTPException(status_code=404, detail="Report not found")
+    return report
+
+def fetch_report_from_db(report_id: int):
+    """
+    Function to fetch a report from the SQLite database.
+    """
+    try:
+        liteConn = sqlite3.connect("database.db") 
```

Comment:
You should remove this line

Reply:
Since we defined liteConn as global now, we do not need to reconnect to the database.